src/Symfony/Component/Security/Core/Authentication/Token/UsernamePasswordToken.php

```diff
@@ -27,7 +27,7 @@ class UsernamePasswordToken extends AbstractToken
      * Constructor.
      *
      * @param string|object            $user        The username (like a nickname, email address, etc.), or a UserInterface instance or an object implementing a __toString method
-     * @param string                   $credentials This usually is the password of the user
+     * @param mixed                    $credentials This usually is the password of the user
```

Comment:
https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.php#L40-L45

In my app it is null in some cases like impersonation.